### DISCUSSION TITLE
[Question] RayCaster: Different height calculation ways bewteen reward func : base_height_rough_l2 and height_scan func in observation

### DISCUSSION THREAD

**Discussion by AlexanderAbernathy** [fan-ziqi/robot_lab#60] [Q&A] [answered]
# Observation func
sensor.data.pos_w[:, 2].unsqueeze(1) - sensor.data.ray_hits_w[..., 2] - offset

# Reward Func
base_height = torch.mean(asset.data.root_pos_w[:, 2].unsqueeze(1) - sensor.data.ray_hits_w[..., 2], dim=1)

Don't know why there is an offset in the default function. Pls have a look and test.

**Comment by AlexanderAbernathy** [comment, marked as answer]
Some discussions in https://github.com/isaac-sim/IsaacLab/discussions/774
